moveit_core/robot_state/test/robot_state_test.cpp

```diff
@@ -567,72 +571,144 @@ TEST_F(OneRobot, FK)
+    // robot_state.reset(new robot_state::RobotState(*robot_state));
+    traj.push_back(robot_state::RobotStatePtr(new robot_state::RobotState(*robot_state)));
+  }
+
+  ja = robot_state->getVariablePosition("joint_a");
+  jc = robot_state->getVariablePosition("joint_c");
+
+  // 4th waypoint with a small jump of 0.01 in revolute joint and prismatic joint. This should not be considered a jump
+  ja = ja - 0.01;
+  robot_state->setVariablePosition("joint_a", ja);
+  jc = jc - 0.01;
+  robot_state->setVariablePosition("joint_c", jc);
+  traj.push_back(robot_state::RobotStatePtr(new robot_state::RobotState(*robot_state)));
+
+  // 5th waypoint with a large jump of 1.01 in first revolute joint
+  ja = ja + 1.01;
+  robot_state->setVariablePosition("joint_a", ja);
+  traj.push_back(robot_state::RobotStatePtr(new robot_state::RobotState(*robot_state)));
+
+  // 6th waypoint with a large jump of 1.01 in first prismatic joint
+  jc = jc + 1.01;
+  robot_state->setVariablePosition("joint_c", jc);
+  traj.push_back(robot_state::RobotStatePtr(new robot_state::RobotState(*robot_state)));
+
+  // 7th waypoint with no jump
+  traj.push_back(robot_state::RobotStatePtr(new robot_state::RobotState(*robot_state)));
+
+  return traj.size();
 }
 
 TEST_F(OneRobot, testAbsoluteJointSpaceJump)
 {
-  const robot_model::JointModelGroup* joint_model_group = robot_model->getJointModelGroup("base_from_base_to_tip");
+  const robot_model::JointModelGroup* joint_model_group = robot_model->getJointModelGroup("base_from_base_to_e");
   std::vector<std::shared_ptr<robot_state::RobotState>> traj;
 
-  // direct call of absolute version
+  // A revolute joint jumps 1.01 at the 5th waypoint and a prismatic joint jumps 1.01 at the 6th waypoint
+  const std::size_t expected_revolute_jump_traj_len = 4;
+  const std::size_t expected_prismatic_jump_traj_len = 5;
+  const std::size_t expected_full_traj_len = 7;
+  const double expected_revolute_jump_fraction =
+      (double)expected_revolute_jump_traj_len / (double)expected_full_traj_len;
+  const double expected_prismatic_jump_fraction =
+      (double)expected_prismatic_jump_traj_len / (double)expected_full_traj_len;
+
+  // Generate a test trajectory
+  std::size_t full_traj_len = generateTestTraj(traj, robot_model, joint_model_group);
+
+  // Test the generateTestTraj still generates a trajectory of length 7
+  EXPECT_EQ(full_traj_len, expected_full_traj_len);  // full traj should be 7 waypoints long
+
+  // Container for results
+  double fraction;
+
+  // Direct call of absolute version
   generateTestTraj(traj, robot_model, joint_model_group);
-  double fraction = robot_state::RobotState::testAbsoluteJointSpaceJump(joint_model_group, traj, 1.0, 1.0);
-  EXPECT_EQ(traj.size(), 3);  // traj should be cut
-  EXPECT_NEAR(fraction, 3. / 4., 0.01);
+  fraction = robot_state::RobotState::testAbsoluteJointSpaceJump(joint_model_group, traj, 1.0, 1.0);
+  EXPECT_EQ(expected_revolute_jump_traj_len, traj.size());  // traj should be cut
```

Comment:
I cannot help, but for me the numbers are still more readable than these lengthy variables (although you already shortened them, which I acknowledge). I guess, that's because it is much easier to spot and compare figures - they nicely pop out -, while all the variables look too similar in the end.
Probably that's a matter of taste. Hence, consider this as a remark to think about. I will not insist in further changes.

Reply:
I see your point but I think it is better this way